### PR TITLE
[chart.js] Add weight property to ChartDataSets interface for Doughnut/Pie charts.

### PR DESCRIPTION
**Pull request by jordanweiler**
The Chart.js Doughnut/Pie chart can accept a `weight` property that will affect the ring weight but it wasn't yet supported by @types/chart.js so I created this PR. I also ran prettier on the index.d.ts file and added those changes to this PR. I put them on a separate commit to separate the actual change from the cosmetic changes.

Please let me know if you would prefer I remove the prettier fixes from this PR and put them on a different PR or just not do them at all.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  * https://www.chartjs.org/docs/latest/charts/doughnut.html
  * https://github.com/chartjs/Chart.js/blob/master/src/controllers/controller.doughnut.js

### CODE REVIEW

File: types/chart.js/index.d.ts
Comment (ricardo-mello) [inline types/chart.js/index.d.ts]: @ElianCordoba should these formatting be reverted?

Comment (ElianCordoba) [reply types/chart.js/index.d.ts]: Yes please, all of them. Beside that the change looks good

**Review by ElianCordoba — CHANGES_REQUESTED** [review]

File: types/chart.js/index.d.ts
Comment (jordanweiler) [reply types/chart.js/index.d.ts]: I've reverted all of those prettier changes on this PR so that only the weight property is on this PR.

Would you like me to add those prettier formatting changes on a separate PR or just ignore them altogether?

**Review by ElianCordoba — APPROVED** [review]
LGTM, thanks!

File: types/chart.js/index.d.ts
Comment (sandersn) [reply types/chart.js/index.d.ts]: As DT maintainer I'd like to see the prettier changes eventually, but I'll defer to @ElianCordoba and @ricardo-mello.

**Pull request merged**
